src/node_api.cc

```diff
@@ -2011,14 +2012,16 @@ napi_status napi_create_reference(napi_env env,
                                   napi_value value,
                                   uint32_t initial_refcount,
                                   napi_ref* result) {
-  NAPI_PREAMBLE(env);
+  // Omit NAPI_PREAMBLE and GET_RETURN_STATUS because V8 calls here cannot throw
+  // JS exceptions.
+  CHECK_ENV(env);
   CHECK_ARG(env, result);
 
   v8impl::Reference* reference = v8impl::Reference::New(
       env, v8impl::V8LocalValueFromJsValue(value), initial_refcount, false);
 
   *result = reinterpret_cast<napi_ref>(reference);
-  return GET_RETURN_STATUS(env);
+  return napi_ok;
```

Comment:
`NAPI_PREAMBLE(env)` clears the last error, whereas `CHECK_ENV(env)` does not. So, the public APIs which do not start with `NAPI_PREAMBLE(env)` leave the last error status as it was if they simply return `napi_ok`. Thus, I was thinking we should change `napi_clear_last_error(env)` to always return `napi_ok` and then change all these `return napi_ok;` statements to `return napi_clear_last_error(env);`